### ISSUE TITLE
Shredder never finds duplicate files

### ISSUE THREAD

**Issue by HaleTom** [sahib/rmlint#386]
Shredder finds no duplicate files. All it ever shows is a tickmark and `Nothing found!`.

I am using default settings.

Purposefully create duplicate files:

mkdir /tmp/shredder
echo foo > /tmp/shredder/1
echo foo > /tmp/shredder/2

I then select the `/tmp` folder and see the following:
![Nothing found!](https://i.imgur.com/HPsR2YD.png)

At the terminal, I see:
```
% rmlint --gui
Could not import colorlog. This is not an error.
You just will not have colors in the logging output...
13:50:50 INFO     [__init__.py: 42] root: Starting up.
13:50:50 INFO     [application.py:102] application: Loading resources from: /usr/lib/python3.7/site-packages/shredder/resources/shredder.gresource
13:50:50 DEBUG    [application.py:148] application: Instancing views.
13:50:50 INFO     [locations.py:343] locations: Loading entries initially
13:50:50 INFO     [editor.py: 52] editor: Using GtkSourceView since we have it.
13:50:50 DEBUG    [application.py:153] application: Done instancing views.

(shredder:15431): Gtk-WARNING **: 13:50:51.194: Failed to measure available space: Error getting filesystem info for /media/backup: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.196: Failed to measure available space: Error getting filesystem info for /media/ssd: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.196: Failed to measure available space: Error getting filesystem info for /media/32GB-sandisk: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.197: Failed to measure available space: Error getting filesystem info for /media/32GB-sandisk: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.198: Failed to measure available space: Error getting filesystem info for /media/ssd: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.292: Failed to measure available space: Error getting filesystem info for /media/1TB-toshiba: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.292: Failed to measure available space: Error getting filesystem info for /media/1TB-toshiba: No such device

(shredder:15431): Gtk-WARNING **: 13:50:51.292: Failed to measure available space: Error getting filesystem info for /media/backup: No such device
/usr/lib/python3.7/site-packages/shredder/runner.py:170: Warning: g_variant_get_boolean: assertion 'g_variant_is_of_type (value, G_VARIANT_TYPE_BOOLEAN)' failed
  cfg.get_boolean('computation-algorithm'))
13:50:55 INFO     [runner.py:216] runner: Running: rmlint --no-with-color -T duplicates --see-symlinks --hidden -c sh:handler=remove --hardlinked --crossdev --size 1-18446744073709551615 --max-depth 512 -o json -c json:oneline /tmp/shredder
13:50:55 INFO     [runner.py:304] runview: `rmlint` finished.
```

Version info:

```% rmlint --gui --version
Could not import colorlog. This is not an error.
You just will not have colors in the logging output...
version 2.9.0 compiled: Aug 25 2019 at [11:36:05] "Odd Olm" (rev unknown)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support
```

**Comment by HaleTom** [comment]
Running just:

```
rmlint --no-with-color -T duplicates --see-symlinks --hidden -c sh:handler=remove --hardlinked --crossdev --size 1-18446744073709551615 --max-depth 512 -o json -c json:oneline /tmp/shredder
```
I get:

```ERROR: cannot parse --size: Max is smaller than min.```

The default is 16 petabytes. If I change only `peta` to `tera` I can't press `Apply` (greyed out), but can after I change the value `16` to some other value.

Where is the config file written? Googling `shredder rmlint configuration file` doesn't turn up anything.

**Comment by sahib** [comment]
Hello @HaleTom,

this seems to be a bug in the size parser of ``rmlint``. Most likely an integer overflow. Maybe the fix is just to change the default.

> Where is the config file written? Googling shredder rmlint configuration file doesn't turn up anything.

It uses ``gsettings``:

```
$ gsettings list-recursively | grep Shredder
```

What version are you running?

**Comment by HaleTom** [comment]
@sahib v2.9.0 as per the output above.

And truncate the max selectable to not have valid user input break things?

Also there's the can't save units change without changing the number issue.

**Comment by sahib** [comment]
> @sahib v2.9.0 as per the output above.

Pardon, I didn't see this.

> And truncate the max selectable to not have valid user input break things?

Who said I would truncate things? I meant the default in the UI which might not be easily handled by ``gsettings`` (at least I recall issues). There is also a bug in ``rmlint``'s size parser that cannot handle this (even though the internal representation is big enough). I will provide a fix soon.

> Also there's the can't save units change without changing the number issue.

I'm aware.

**Comment by sahib** [comment]
This issue should be resolved in the ``develop`` branch by those commits:

* 0b22580b ui: fix bug when changing the size factor did not let you apply settings
* f62c2b87 fix handling of very large sizes in --size parsing (now up to 2*64 properly supported)

Make sure to install ``rmlint`` when trying out. The UI will take the ``rmlint`` from your ``PATH``.

**Comment by sahib** [comment]
Closing due to lack of activity. Please re-open if problem persists.

**Comment by verajosemanuel** [comment]
same issue here. Fresh install using defaults copied a file twice but nothing is found,
Ubuntu Focal Fossa , shredder 2.9.0

**Comment by sahib** [comment]
@verajosemanuel ...and did you try latest ``develop``?

**Comment by verajosemanuel** [comment]
Removed package, installed using goit develop version. Solved!
Thanks

**Comment by NicoWeio** [comment]
Using the packaged version (_rmlint-gui_ 2.9.0-2 / focal) worked for me **after lowering the upper file size limit** from the default value to (in my case) 100 GB. Maximum working value not tested. That might be an alternative for folks who want to stick with their distribution's provided package version.

**Comment by tony-travis** [comment]
I've just reported this as a BUG in the version packaged for Ubuntu 20.04 LTS
https://bugs.launchpad.net/ubuntu/+source/rmlint/+bug/1951596

The problem is that the default 16 PB max file size is interpreted as a negative number.

This is the error running rmlint with the default 16 PB value:

ajt@pilot:~$ rmlint --no-with-color -T duplicates --see-symlinks --partial-hidden -c sh:handler=remove --hardlinked --crossdev --size 1-18446744073709551615 --max-depth 512 -o json -c json:oneline /home/ajt/Documents/Work/Minke
ERROR: cannot parse --size: Max is smaller than min.

**Comment by minuxlintebiandedition** [comment]
I have just installed rmlint and ran it on a set of mostly music and video files with lots of known duplicates.

Nothing found, nothing done, 2025, bookworm version.

**Comment by ouinouin** [comment]
got the issue with the 2.9.0 from linux mint packages , i fixed the issue by lowering the max filesize effectively

**Comment by RayOei** [comment]
> I have just installed rmlint and ran it on a set of mostly music and video files with lots of known duplicates.
> 
> Nothing found, nothing done, 2025, bookworm version.

Please add more information and versions used. As it stands this should be solved in `2.10.3`
